### ISSUE TITLE
Stable Baselines uses the forward method in the base features extractor class instead of my custom one

### ISSUE THREAD

**Issue by RobertHua96** [DLR-RM/stable-baselines3#616]
Hi there, I implemented a custom feature extractor for images using a vision transformer. When that model is being used in 'model.learn' I get a not implemented error for the forward method - which is coming from the BaseFeaturesExtractor class. I'm wondering why the parent class is being accessed instead of my child class which has the implemented forward function? 

### Additional context
My code:

`
!pip install nystrom-attention
!pip install vit-pytorch
import torch
from torch import nn
from torchvision import datasets, transforms
from vit_pytorch import ViT
from vit_pytorch.efficient import ViT
from nystrom_attention import Nystromformer
from einops import rearrange, repeat
from einops.layers.torch import Rearrange
class CustomViT(BaseFeaturesExtractor):
  def __init__(self, observation_space: gym.spaces.Box, features_dim: int = 4096):
    super(CustomViT, self).__init__(observation_space, features_dim)
    #We assume CxHxW (channels first)
    n_input_channels = observation_space.shape[0]
    self._observation_space = observation_space
    
    self.efficient_transformer = Nystromformer(
    dim = 4096,
    depth = 12,
    heads = 8,
    num_landmarks = 256
    )
    
    self.vit = Features_ViT(
    dim = 4096,
    image_size = 1024,
    patch_size = 32,
    num_classes = 1000,
    transformer = self.efficient_transformer
    )

    def forward(self, observations: torch.Tensor) -> torch.Tensor:
      return self.vit.forward(self.img_to_tensor(observations))

  @property
  def features_dim(self) -> int:
      self._features_dim = 4096
      return self._features_dim

  def img_to_tensor(obs):
    img_tensor = transforms.ToTensor(obs)
    img_tensor = img_tensor.unsqueeze(0)
    return img_tensor`

`class Features_ViT(nn.Module):
    def __init__(self, *, image_size, patch_size, num_classes, dim, transformer, pool = 'cls', channels = 4):
        super().__init__()
        assert image_size % patch_size == 0, 'image dimensions must be divisible by the patch size'
        assert pool in {'cls', 'mean'}, 'pool type must be either cls (cls token) or mean (mean pooling)'
        num_patches = (image_size // patch_size) ** 2
        patch_dim = channels * patch_size ** 2

        self.to_patch_embedding = nn.Sequential(
            Rearrange('b c (h p1) (w p2) -> b (h w) (p1 p2 c)', p1 = patch_size, p2 = patch_size),
            nn.Linear(patch_dim, dim),
        )

        self.pos_embedding = nn.Parameter(torch.randn(1, num_patches + 1, dim))
        self.cls_token = nn.Parameter(torch.randn(1, 1, dim))
        self.transformer = transformer

        self.pool = pool
        self.to_latent = nn.Identity()

        self.mlp_head = nn.Sequential(
            nn.LayerNorm(dim),
            nn.Linear(dim, num_classes)
        )

    def forward(self, img):
        x = self.to_patch_embedding(img)
        b, n, _ = x.shape

        cls_tokens = repeat(self.cls_token, '() n d -> b n d', b = b)
        x = torch.cat((cls_tokens, x), dim=1)
        x += self.pos_embedding[:, :(n + 1)]
        x = self.transformer(x)

        x = x.mean(dim = 1) if self.pool == 'mean' else x[:, 0]

        x = self.to_latent(x)
        return x`

### Checklist

- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)

**Comment by araffin** [comment]
Hello,
please fill the issue template completely and provide a *minimal working* example to reproduce the issue (and please use markdown codeblock, see example in issue template).
